src/helper/subscriber/waf.cpp

```diff
@@ -123,11 +129,12 @@ void extract_tags_and_metrics(parameter_view diagnostics, std::string &version,
 
             it = rules.find("errors");
             if (it != rules.end()) {
-                meta[tag::event_rules_errors] = parameter_to_json(it->second);
+                meta[std::string(tag::event_rules_errors)] =
```

Comment:
To avoid having to generate a string, you can use heterogeneous lookups, by changing the map to the following:

```
std::map<std::string, std::string, std::less<>> &meta
```

Reply:
Turns out this isn't supported for `operator[]` until C++23 (as far as I can tell), an alternative could be:

```
if (auto it = meta.find(tag::event_rules_errors); it != meta.end()) {
    it->second = parameter_to_json(it->second);
} else {
    meta.emplace(std::string(tag::event_rules_errors), parameter_to_json(it->second));
}
```

But to be honest, those tags are not going to be in meta anyway so it might not be worth changing.

Reply:
Saying that, can't get it to work with C++23 so I'm not sure if it made it.

Reply:
🤯